app/models/event_date.rb

```diff
@@ -21,4 +21,9 @@ class EventDate < ApplicationRecord
   scope :future, lambda {
     where('event_date_key >= ?', Date.today.to_s.delete('-'))
   }
+
+  scope :end_datetime, lambda {
```

Comment:
Should this query just be included as part of the active scope? In my mind an event is no longer "active" after it's published_end_datetime.

Reply:
I can't think of cases where the coupling of the status_id and published_end_datetime would cause problems but that may be a lack of imagination.  I can see a case where a user might want to see their history of registrations but I am guessing that would come from an interface with the panty trak db.  

@davidpickering do you have any objections to combining these scopes?  I successfully tested it in my local environment.

Reply:
I cannot see any problems with including published_end_datetime as part of the active scope. That extends this feature throughout all of the default uses of this data, and we still have flexibility to request by another scope if need be. So yes, default behavior should be to 'hide' the ones that we do not want to be published. If we build an admin interface, we'll build the way to get to this data.

Reply:
> I cannot see any problems with including published_end_datetime as part of the active scope. That extends this feature throughout all of the default uses of this data, and we still have flexibility to request by another scope if need be. So yes, default behavior should be to 'hide' the ones that we do not want to be published. If we build an admin interface, we'll build the way to get to this data.

Here is how I coded the combined active scope:

scope :active, -> { where('event_dates.status_id = 1 AND ? <= published_end_datetime',
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;DateTime.current.utc.strftime('%Y-%m-%d %H:%M:%S'))